Tasks/DownloadBuildArtifact/FileContainer.ts

```diff
@@ -0,0 +1,80 @@
+import * as tl from 'vsts-task-lib/task';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import {BuildArtifact} from 'vso-node-api/interfaces/BuildInterfaces';
+import {FileContainerItem, ContainerItemType} from 'vso-node-api/interfaces/FileContainerInterfaces';
+import {IFileContainerApi} from 'vso-node-api/FileContainerApi';
+import {WebApi, getBearerHandler} from 'vso-node-api/WebApi';
+
+import {DownloadItem, download} from './Downloader';
+import {ArtifactProvider} from './ArtifactProvider';
+
+export class FileContainerProvider implements ArtifactProvider {
+    public supportsArtifactType(artifactType: string): boolean {
+        return !!artifactType && artifactType.toLowerCase() === "container";
+    }
+
+    public async downloadArtifact(artifact: BuildArtifact, targetPath: string): Promise<void> {
+        if (!artifact || !artifact.resource || !artifact.resource.data) {
+            throw new Error(tl.loc("FileContainerInvalidArtifact"));
+        }
+
+        let containerParts: string[] = artifact.resource.data.split('/', 3);
+        if (containerParts.length !== 3) {
+            throw new Error(tl.loc("FileContainerInvalidArtifactData"));
+        }
+
+        let containerId: number = parseInt(containerParts[1]);
+        let containerPath: string = containerParts[2];
+
+        let accessToken = getAuthToken();
+        let credentialHandler = getBearerHandler(accessToken);
+        let collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+        let vssConnection = new WebApi(collectionUrl, credentialHandler);
+
+        let fileContainerApi = vssConnection.getFileContainerApi();
+
+        // get all items
+        let items: FileContainerItem[] = await fileContainerApi.getItems(containerId, null, containerPath, false, null, null, false, false);
+
+        // ignore folders
+        items = items.filter(item => item.itemType === ContainerItemType.File);
+        tl.debug(`Found ${items.length} File items in #/${containerId}/${containerPath}`);
```

Comment:
Yes, intentional, as I'm referring to the itemType value.  Items are "File" or "Folder", IIRC.  Although I suppose that value isn't relevant to your average build watcher...